File: src/test/java/org/effrafax/spa/csp/steps/CspSteps.java
package org.effrafax.spa.csp.steps;

import cucumber.api.DataTable;
import cucumber.api.java.en.Given;
import cucumber.api.java.en.Then;
import cucumber.api.java.en.When;
import cucumber.runtime.PendingException;

public class CspSteps {
	@Given("^I have a CSP problem$")
	public void I_have_a_CSP_problem() throws Throwable {
	    // Express the Regexp above with the code you wish you had
	    throw new PendingException();
	}

	@Given("^I have a variable '\\w+' with domain \\[(.*)\\]$")
	public void I_have_a_variable_X_with_domain_(String variable, String domain) throws Throwable {
	    // Express the Regexp above with the code you wish you had
	    throw new PendingException();
	}

	@Given("^I have a constraint: (.*)$")
	public void I_have_a_constraint(String constraint) throws Throwable {
		// Express the Regexp above with the code you wish you had
		throw new PendingException();
	}

	@When("^I solve the problem$")
	public void I_solve_the_problem() throws Throwable {
	    // Express the Regexp above with the code you wish you had
	    throw new PendingException();
	}

	@Then("^I expect solutions$")
	public void I_expect_solutions(DataTable table) throws Throwable {
	    // Express the Regexp above with the code you wish you had
	    // For automatic conversion, change DataTable to List<YourType>
	    throw new PendingException();
	}
}

File: features/steps/csp_steps.py
from lettuce import step

@step(u'Given I have a CSP problem')
def given_i_have_a_csp_problem(step):
    assert False, 'This step must be implemented'

@step(u'And I have a variable \'([^\']*)\' with domain \[([^\]]*)\]')
def and_i_have_a_variable_group1_with_domain_0_1_2(step, variable, domain):
    assert False, 'This step must be implemented'

@step(u'And I have a constraint: (.*)')
def and_i_have_a_constraint_x_y(step, constraint):
    assert False, 'This step must be implemented'

@step(u'When I solve the problem')
def when_i_solve_the_problem(step):
    assert False, 'This step must be implemented'

@step(u'Then I expect solutions')
def then_i_expect_solutions(step):
    assert False, 'This step must be implemented'
